test/e2e/data/infrastructure-azure/v1beta1/bases/mp.yaml

```diff
@@ -29,7 +29,6 @@ metadata:
   name: ${CLUSTER_NAME}-mp-0
   namespace: default
 spec:
-  identity: SystemAssigned
```

Comment:
This base is leftover from this old PR: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1423/files#diff-64ec3c60700d0c80917da2dc8f447b2cc531a482c32abdef2466c7bb7e9740d7R32

Current prow mp templates do not have this field: https://github.com/willie-yao/cluster-api-provider-azure/blob/357145797da465de3ba5d55f31a6e7d85e71bb7f/templates/cluster-template-machinepool.yaml#L161